### DISCUSSION TITLE
[Linux] Use a script that updates the metainfo.xml with the changelog before the release is tagged

### DISCUSSION THREAD

**Discussion by Ferenc-** [organicmaps/organicmaps#4699] [Ideas]
**Problem:**
The lack of timely changelog update in `app.organicmaps.desktop.metainfo.xml`, before the release is tagged,
is a source of a couple of issues:
1. It generates recurring extra effort for project memebers to write and review the changelog update with a PR.
2. It also generates extra work for packagers like the flatpak maintainers, to[ patch the changelog ](https://github.com/flathub/app.organicmaps.desktop/blob/8e2b573a713abc96413f9f2710b6e41369ac6fd3/changelog-2023.02.13-11.patch) on top of the tagged release.
3. Such patches make the linux version considered `dirty`, which degrades the quality as pointed out by @biodranik [here](https://github.com/organicmaps/organicmaps/issues/4668#issuecomment-1454726519).

As @rtsisyk has also [pointed out here](https://github.com/organicmaps/organicmaps/pull/4475#issuecomment-1428565520), we should implement a software tool to update that.

I would be willing to implement such a tool, and before any work could begin:

1. There should be an agreement on the consistent usage of it, because if nobody will use it for the intended purpose, then it is a waste of time to implement it. From all those project members, who have the privilege to create a release, and decide on what to write into the changelogs, we would need an OK on this. Based on the current `GOVERNANCE.md`, I would like to ask the board members, to express their opinion on this. (@biodranik, @vng, @rtsisyk)

2. Once all the involved parties agree, that it makes sense to supply the changelog input to a tool and then tag the release,
then we could collect some requirements for such a tool.
2.1 What OS should such a tool support? Is it OK, to cover MacOS, Linux and WSL? Or do we have to cover more?
2.2 What dependencies such a tool allowed to have? C++ compiler? Bash? Some xml tools? Maybe the appstream cli or Qt lb? Python?
2.3 How would such a tool get the changelog input? Interactively typed in on the cli? Or in a GUI window? cli arguments? File? Fetching from an external resource?

**Comment by biodranik** [comment]
We create release notes PR now before iOS and Android release. It is possible to create another xml file in some semi-automated way at this moment. If it was a text file like we are using for iOS, then it would be very easy. 

However, a longer term solution would somehow automate release notes and their translations for all platforms without creating a PR manually and then reading/merging all translators' suggestions. Maybe a simple browser-based tool like Weblate, or self-made one, can simplify the process.

Issues:
- Whats new have different chars count for different platforms. 
- Some of them may not be translated, but should be updated at least in English. 
- Something can be changed before the release.

**Reply by Ferenc-** [reply]
So do I understand it correctly, that currently there is no interest in a tool, that only focuses on `app.organicmaps.desktop.metainfo.xml`?
And on the short term, the only way is to be available when the release notes PR is open, and contribute.
And on the long term, the automation tool would cover more than just `app.organicmaps.desktop.metainfo.xml`

**Reply by biodranik** [reply]
Anything worth automation should be automated )

**Reply by Ferenc-** [reply]
So is that a yes then? Then what about the requirements mentioned in the `2`. point [here](https://github.com/orgs/organicmaps/discussions/4699#discussion-4929208)?

**Reply by biodranik** [reply]
The issue is only with formatting. A bash or python script can help a bit. 

Full Automation is not possible because the input for desktop releases is different from iOS and Android.
